broadbean/sequence.py

```diff
@@ -478,21 +481,102 @@ def description(self):
         """
         desc = {}
 
-        for pos, elem in self._data.items():
+        for pos, elem in self._data.items():           
             desc[str(pos)] = {}
             desc[str(pos)]['channels'] = elem.description
             try:
                 sequencing = self._sequencing[pos]
-                seqdict = {'Wait trigger': sequencing[0],
-                           'Repeat': sequencing[1],
-                           'Event jump to': sequencing[2],
-                           'Go to': sequencing[3]}
+                seqdict = {'Wait trigger': sequencing['twait'],
+                           'Repeat': sequencing['nrep'],
+                           'jump_input': sequencing['jump_input'],
+                           'jump_target': sequencing['jump_target'],
+                           'Go to': sequencing['goto']}
                 desc[str(pos)]['sequencing'] = seqdict
             except KeyError:
                 desc[str(pos)]['sequencing'] = 'Not set'
-
+        desc['awgspecs'] = self._awgspecs
         return desc
 
+    def write_to_json(self, path_to_file: str) -> None:
```

Comment:
I would propose to have two levels of functions on the `Sequence` object. One for doing `Seq -> JSON str` and another for doing `JSON str -> file` (and similarly in the other direction).

Reply:
We now have a two level process one for doing or reading a dict and a other to read or write the dict to a file. However, there is no separate function for making the dict into a json string.

Reply:
Hmm... I guess I can live with that. I somehow dislike having the file I/O living on the class, but that's probably just an irrational taste. As long as the object can be constructed from a dict, I guess the rest is details.